npe2/manifest/_npe1_shim.py

```diff
@@ -0,0 +1,45 @@
+import logging
+import warnings
+
+from .._from_npe1 import manifest_from_npe1
+from .package_metadata import PackageMetadata
+from .schema import PluginManifest, discovery_blocked
+
+try:
+    from importlib import metadata
+except ImportError:
+    import importlib_metadata as metadata  # type: ignore
+
+
+logger = logging.getLogger(__name__)
+
+
+class NPE1Shim(PluginManifest):
```

Comment:
A comment introducing what the general motivation and approach behind the shim is might go here (or maybe module level).

Reply:
also, describe how the general inspection works and document it somewhere clearly